hydra/_internal/core_plugins/importlib_resources_config_source.py

```diff
@@ -40,23 +43,33 @@ def load_config(
         if not res.exists():
             raise ConfigLoadError(f"Config not found : {normalized_config_path}")
 
-        with res.open(encoding="utf-8") as f:
-            header_text = f.read(512)
-            header = ConfigSource._get_header_dict(header_text)
-            self._update_package_in_header(
-                header=header,
-                normalized_config_path=normalized_config_path,
-                is_primary_config=is_primary_config,
-                package_override=package_override,
-            )
-            f.seek(0)
-            cfg = OmegaConf.load(f)
-            return ConfigResult(
-                config=self._embed_config(cfg, header["package"]),
-                path=f"{self.scheme()}://{self.path}",
-                provider=self.provider,
-                header=header,
-            )
+        vi = sys.version_info
+        python_version = f"{vi.major}.{vi.minor}.{vi.micro}"
+        zipfile_path = parse_version(python_version) >= parse_version(
+            "3.8"
+        ) and isinstance(res, zipfile.Path)
+
+        f = res.open() if zipfile_path else res.open(encoding="utf-8")
```

Comment:
I prefer
```
if foo:
  res = ...
else:
  res = ...
```
in this case as it makes it more obvious what is happening.
It will also allow you to place additional logic in each branch.